db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+. Specify the database connection information and click *OK*.
+. Set the *SQL Query Text* field to the SQL query to execute, for example `SELECT * FROM PLANET WHERE name = :name`.
+
+The following screenshot shows the *Select* operation configuration:
+
+.Select operation configuration
+image::database-select-operation-1.png[Select operation configuration in Studio]
+
+In the XML editor, the `<db:select>` configuration looks like this:
+[source,xml,linenums]
+----
+<db:select config-ref="dbConfig">
+  <db:sql>SELECT * FROM PLANET WHERE name = :name</db:sql>
+</db:select>
+----
+
+== Configure the Input Parameters Field
+
+To protect database queries, configure the *Input Parameters* field in the Select operation by adding variable values to the SQL statement you execute in the database. The primary goal of the Select operation is to supply a SQL query and use DataWeave for the parameters.
+
+The parameters you provide are a map in which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix, for example `where id = :myParamName`. The map’s values contains the actual assignation for each parameter.
+
+The advantages of using input parameters to configure the `WHERE` clause in a SQL `SELECT` statement are that it makes the query immune to SQL injection attacks and enables optimizations that are not possible otherwise improving the application performance.
+
+For security reasons, do not directly write `<db:sql>SELECT * FROM PLANET WHERE name = #[payload] </db:sql>`.
+
+DataSense is available for the operation's input and output. Database Connector analyzes the query and automatically calculates the structure of the query's output by analyzing the projection section of the SQL statement. At the same time, by comparing the conditions in the `WHERE` clause to the table structure, it generates DataSense input to help you build the DataWeave script that generates the input parameters.
+
+In the following example, you supply input parameters as key-value pairs embedding a DataWeave script. The keys use the colon character (`:`) to reference a parameter value by name:
+
+. In your Studio flow, select the *Select* operation.
+. In the operation configuration screen, set the *SQL Query Text* field to `SELECT * FROM PLANET WHERE name = :name`.
+. Set the *Input Parameters* field to `{'name' : payload}`.
+
+The following screenshot shows the *Input Parameters*  configuration:
+
+.Input Parameters configuration
+image::database-select-operation-2.png[Input Parameters configuration in Studio]
+
+In the XML editor, the `<db:input-parameters>` configuration looks like this:
+[source,xml,linenums]
+----
+<flow name="DatabaseFlow">
+  <db:select config-ref="dbConfig">
+    <db:sql>SELECT * FROM PLANET WHERE name = :name</db:sql>
+    <db:input-parameters>
+      #[{'name' : payload}]
+    </db:input-parameters>
+  </db:select>
+</flow>
+----
+
```

Comment:
Instead of NOTEs, I suggest that the following be converted to text about version-specific considerations: perhaps an intro statement and then a two-item definition list? Avoid using boldface to create narrative headings.

Reply:
I also wonder if these NOTE texts should appear before the task, kind of like prerequisites or requirements?

Reply:
NOTES copyedited only; consider alternatives above, or other ways to insert this info into main narrative instead of tacking it onto the end of the task.